src/timer/timer_game_calendar.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file timer_game_calendar.h Definition of the game-calendar-timer */
+
+#ifndef TIMER_GAME_CALENDAR_H
+#define TIMER_GAME_CALENDAR_H
+
+/**
+ * Timer that is increased every 27ms, and counts towards ticks / days / months / years.
+ *
+ * The amount of days in a month depends on the month and year (leap-years).
+ * There are always 74 ticks in a day (and with 27ms, this makes 1 day 1.998 seconds).
+ *
+ * IntervalTimer and TimeoutTimer based on this Timer are a bit unusual, as their count is always one.
+ * You create those timers based on a transition: a new tick, a new day, a new month or a new year.
+ *
+ * Additionally, you need to set a priority. To ensure deterministic behaviour, events are executed
+ * in priority. It is important that if you assign NONE, you do not use Random() in your callback.
+ * Other than that, make sure you only set one callback per priority.
```

Comment:
This is not specific to Random().
All non-GUI-code handlers should be sorted, so all priorities should have a single handler only.
Currently Priority::ENGINE is used twice, and I can imagine that already desyncs. Engine and track-types become available depending on each other, ...

Reply:
They aren't used twice. Well, strictly seen, but one day on Day, and one time on Month. That order is well defined. So it is deterministic currently, and I wouldn't see where any desync would be possible currently :)

You overlooked that they are on different triggers, or am I missing something?

Reply:
Ah yes, I missed that Day/Month thing. Can there be an assertion in the gameloop timer, that the handlers with priority are strictly ordered?

Reply:
can do!

I also planned to do it, but forgot .. so tnx for pointing out it was missing :D

As if this fails means that a developer made a big boo-boo, the code only runs for assert-enabled builds. Means I can just `assert` to detect issues :)